code/src/components/MovieList.js

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useState } from "react";
+import { Link, useHistory } from "react-router-dom";
+import moment from "moment";
+//three api end points
+import { API_MOVIE_LIST_POPULAR } from "./Links";
+import { API_MOVIE_LIST_TOP_RATED } from "./Links";
+import { API_MOVIE_LIST_UPCOMING } from "./Links";
+import { OverLay } from "./OverLay";
+import { Select } from "./Select";
+import { Loader } from "./Loader";
+import { RandomButton } from "./RandomButton";
+import "../index.css";
+
+export const MovieList = () => {
+  const [movies, setMovies] = useState([]);
+  const [loading, setLoading] = useState(true);
+  // special state to hold current state of the select
+  const [change, changeTracker] = useState("popular");
+  //use history is used to forward user to the random movie page, connecting it to the movie details route with use of push
+  const history = useHistory();
+
+  //Thats the function that perform fetch on click and forward user to the random movie page
+  const onRandomButtonClick = () => {
+    //picking a random page out of top rated movie API end point
+    const randomNumber = Math.floor(Math.random() * 450);
+    const url = `https://api.themoviedb.org/3/movie/top_rated?api_key=4cfc03c4e32397dfd5018e9bb30b640c&language=en-US&page=${randomNumber}`;
+    fetch(url)
+      .then((res) => res.json())
+      .then((json) => {
+        //picking a random movie out of the json that we fetched
+        const randomMovie = Math.floor(Math.random() * json.results.length);
+        // picking an id of the random movie we fetched and selected above
+        const randomMovieID = json.results[randomMovie].id;
+        // redirecting the user to the correct routing path
+        history.push(`/movieDetails/${randomMovieID}`);
+      });
+  };
```

Comment:
Really cool function that you added here! Feels like something you could see in netflix!